### ISSUE TITLE
[Webpack] Cannot resolve module 'package.version'

### ISSUE THREAD

**Issue by ericclemmons** [auth0/lock#302]
I'm using the code from #300 and even tried the README on `master`, but having major issues:

```
ERROR in ./src/~/auth0-lock/index.js
Module not found: Error: Cannot resolve module 'package.version' in /Users/Eric/Projects/HigherEducation/platform/src/node_modules/auth0-lock
 @ ./src/~/auth0-lock/index.js 114:20-46

ERROR in ./src/~/auth0-lock/lib/html/main.ejs
Module parse failed: /Users/Eric/Projects/HigherEducation/platform/src/node_modules/auth0-lock/lib/html/main.ejs Line 1: Unexpected token <
You may need an appropriate loader to handle this file type.
| <div id="a0-lock" class="a0-lock<%= (locals && locals.options.theme) ? ' a0-theme-' + locals.options.theme : ''%>" dir="auto">
|   <div class="a0-signin">
|     <div class="a0-popup <%= (locals && locals.cordova) ? 'a0-cordova' : '' %> <%= (locals && locals.ios) ? 'a0-ios' : '' %>">
 @ ./src/~/auth0-lock/index.js 20:15-45

(This happens x10 times!)

ERROR in ./src/~/auth0-lock/lib/i18n/dicts.js
Module not found: Error: Cannot resolve module 'json-loader' in /Users/Eric/Projects/HigherEducation/platform/src/node_modules/auth0-lock/lib/i18n
 @ ./src/~/auth0-lock/lib/i18n/dicts.js 2:8-37

(This happens x100 times!)
```

I even tried:

```js
    resolveLoader: {
      moduleDirectories: ["node_modules"],
      packageMains: ["loader", "main"],
      root: path.join(context, "node_modules"),
    },
```

and

```js
      {
        include: "node_modules/auth0-lock",
        loaders: [
          "transform-loader/cacheable?brfs",
          "transform-loader/cacheable?packageify",
        ],
        test: /\.js$/,
      },
      {
        include: "node_modules/auth0-lock",
        loader: "transform-loader/cacheable?ejsify",
        test: /\.ejs$/,
      },
      {
        include: ["node_modules/auth0-js", "node_modules/auth0-lock"],
        loader: "json-loader",
        test: /\.json$/,
      },
```

I think I might have to just publish the compiled version at <https://cdn.auth0.com/js/lock-8.2.min.js> to NPM under another name so that way it's actually `npm install`able :(

**Comment by gnandretta** [comment]
@ericclemmons what version of npm are you using?

**Comment by ericclemmons** [comment]
@gnandretta 

```shell
$  node --version && npm --version
v5.1.0
3.5.2
```

Managed via my `.nvmrc` file.

(I ended up getting the project working with React by creating a custom component for it, but that doesn't fix the underlying problem.)

Let me know if you need anything else.

**Comment by maoueh** [comment]
Hi @ericclemmons 

PR #300 would work only with `<= 8.1.5` and `>= 8.2.1` (broken in `8.2.0`). Can you share the exact steps you have used to test the example. I'm wondering about the line:

```
Cannot resolve module 'json-loader'
```

Which seems to indicate missing dependencies. What is present in the `node_modules` folder (`ls -l node_modules` from the root directory of the example?

**Comment by maoueh** [comment]
In fact, sorry for the noise, I just saw that they changed the actual way to load `auth0` with webpack. Will need to revisit my PR #300 I think.

**Comment by ericclemmons** [comment]
Oh, I had `8.2.0`!  And apparently `8.2.1` just landed :D

Let me setup a small example with the test-case & either close or ping you guys.

(P.S. `json-loader` exists & is correctly parsing `package.json` & other `.json` files within my project)

**Comment by maoueh** [comment]
@gnandretta I had tested using the old examples, will test with the new way of doing things. At least the old way works :)

**Comment by maoueh** [comment]
@gnandretta Using my PR #300 and version `8.2.1`, the examples work. But using the examples straight from the `master` branch (currently 4742f4c), the example fails with:

```
ERROR in ./~/auth0-lock/lib/mode-kerberos/loggedin.ejs
Module parse failed: /work/auth0/lock/examples/webpack/node_modules/auth0-lock/lib/mode-kerberos/loggedin.ejs Line 1: Unexpected token <

(Many mores)
```

**Comment by gnandretta** [comment]
@ericclemmons there's a known issue with npm 3, see [this comment](https://github.com/auth0/lock/issues/225#issuecomment-173244897). It looks like a fix (npm/npm#10185) is already on the way.

@maoueh I believe you're having a different issue, are you using Windows right? that would explain why the current example doesn't work but the one in your PR does. Let's continue the discussion on #300.

**Comment by ericclemmons** [comment]
I'll close this until the fix lands & we'll go there...
